### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by krisbitney**
This PR will resolve all of the newly created issues that are assigned to me.

### CODE REVIEW

**Comment by sirpy** [comment]
@L03TJ3 please review

**Comment by L03TJ3** [comment]
@krisbitney could you in the description list with # <issue> which ones exactly are fixed

File: packages/app/src/components/DonateComponent.tsx [hunk 1]
```diff
@@ -443,26 +447,38 @@ function DonateComponent({ collective }: DonateComponentProps) {
             !!(chain?.id && chain.id in SupportedNetwork),
             isInsufficientLiquidity,
             isUnacceptablePriceImpact,
-            isInsufficientBalance
+            isInsufficientBalance,
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/components/DonateComponent.tsx]: https://github.com/GoodDollar/GoodCollective/pull/111/commits/d48eb5399a49751c9370097f854832d05640caec

https://github.com/GoodDollar/GoodCollective/pull/111/commits/872f831435bf9546f40ff6c39f4fb5412453f9a4

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

File: packages/app/src/components/DonateComponent.tsx [hunk 2]
```diff
@@ -62,22 +53,35 @@ function DonateComponent({ collective }: DonateComponentProps) {
...
 
+  const tokenList = useTokenList();
+  const isOneTime = frequency === Frequency.OneTime;
+  const currencyOptions: { value: string; label: string }[] = useMemo(() => {
+    let options = Object.keys(tokenList).map((key) => ({
+      value: key,
+      label: key,
+    }));
+    if (isOneTime) {
+      options = [options.find((option) => option.value === 'G$')!];
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/DonateComponent.tsx:59]: it is an array

File: packages/app/src/hooks/useRecentTransactions.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Transaction } from '../models/models';
+
+export const useRecentTransactions = (collective: `0x${string}`, maxN: number): Transaction[] => {
+  return Array(maxN).map((e, i) => ({
+    hash: '0x123' + i.toString(),
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/hooks/useRecentTransactions.ts:5]: mock data

File: packages/app/src/hooks/useContractCalls/useSupportFlow.ts [hunk 4]
```diff
@@ -52,7 +52,11 @@ export function useSupportFlow(
       return;
     } catch (error) {
       toggleCompleteDonationModal(false);
-      onError(`An unexpected error occurred: ${error}`);
+      console.error(error);
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/hooks/useContractCalls/useSupportFlow.ts:55]: No, it is not. Please review the requirements here: https://serota.notion.site/GoodCollective-dApp-MVP-RFP-ba06e28f43e24b85aa336ad7f983c69f

File: packages/app/src/components/TransactionList/TransactionListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    ? formatAddress(userAddress)
+    : '0x';
+
+  const formattedAmount: string = new Decimal(ethers.utils.formatEther(rawAmount) ?? 0).toString();
+  const formattedFee: string = new Decimal(ethers.utils.formatEther(fee) ?? 0).toString();
+
+  return (
+    <View style={styles.row}>
+      {donation ? (
+        <View style={[styles.bar, { backgroundColor: Colors.green[100] }]} />
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/TransactionList/TransactionListItem.tsx:40]: No, it creates a colored bar

File: packages/app/src/components/Breadcrumb.tsx [hunk 6]
```diff
@@ -3,18 +3,34 @@ import { Colors } from '../utils/colors';
...
+function Breadcrumb({ path }: BreadcrumbProps) {
   const navigate = useNavigate();
+
   return (
-    <TouchableOpacity style={styles.container} onPress={() => navigate(-1)}>
-      <Image source={chevronRight} style={styles.backIcon} />
-      <Text style={styles.activePage}>{previousPage ?? 'GoodCollective Home'}</Text>
-      <Text style={styles.nextPage}> / {currentPage}</Text>
+    <TouchableOpacity style={styles.container}>
+      <TouchableOpacity onPress={() => navigate(-1)}>
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/Breadcrumb.tsx]: I fixed this in a new commit. However, I created a constant. I don't understand the callback you're requesting--where does `event` come from? And I'm not sure it makes sense to memoize this.

File: packages/app/src/components/CompleteDonationModal.tsx [hunk 7]
```diff
@@ -18,11 +18,7 @@ const CompleteDonationModal = ({ openModal, setOpenModal }: CompleteDonationModa
         <View style={styles.centeredView}>
           <View style={styles.modalView}>
             <View style={modalStyles.modalCloseIconWrapper}>
-              <TouchableOpacity
-                style={modalStyles.modalCloseIcon}
-                onPress={() => {
-                  setOpenModal(false);
-                }}>
+              <TouchableOpacity style={modalStyles.modalCloseIcon} onPress={() => setOpenModal(false)}>
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/CompleteDonationModal.tsx]: fixed in a new commit

File: packages/app/src/components/DonateComponent.tsx [hunk 8]
```diff
@@ -115,38 +117,26 @@ function DonateComponent({ collective }: DonateComponentProps) {
...
 
   const totalDecimalDonation = new Decimal(duration * decimalDonationAmount);
   const totalDonationFormatted = totalDecimalDonation.toDecimalPlaces(currencyDecimals, Decimal.ROUND_DOWN).toString();
 
-  const isInsufficientBalance = donorCurrencyBalance ? totalDecimalDonation.gt(donorCurrencyBalance) : true;
-  const isInsufficientLiquidity = currency !== 'G$' && swapRouteStatus !== SwapRouteState.READY;
-  const isUnacceptablePriceImpact = currency !== 'G$' && priceImpact ? priceImpact > acceptablePriceImpact : false;
+  const isNonZeroDonation = totalDecimalDonation.gt(0);
+  const isInsufficientBalance =
+    isNonZeroDonation && (donorCurrencyBalance ? totalDecimalDonation.gt(donorCurrencyBalance) : true);
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/DonateComponent.tsx]: fixed in a new commit

File: packages/app/src/components/DonateComponent.tsx [hunk 9]
```diff
@@ -231,7 +221,13 @@ function DonateComponent({ collective }: DonateComponentProps) {
             </View>
             <Dropdown
               value={frequency}
-              onSelect={(value: string) => setFrequency(value as Frequency)}
+              onSelect={(value: string) => {
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/DonateComponent.tsx]: fixed in a new commit

File: packages/app/src/components/DonateComponent.tsx [hunk 10]
```diff
@@ -264,7 +260,13 @@ function DonateComponent({ collective }: DonateComponentProps) {
             <>
               <Dropdown
                 value={frequency}
-                onSelect={(value: string) => setFrequency(value as Frequency)}
+                onSelect={(value: string) => {
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/DonateComponent.tsx]: fixed in a new commit

File: packages/app/src/components/ErrorModal.tsx [hunk 11]
```diff
@@ -16,7 +16,7 @@ const ErrorModal = ({ openModal, setOpenModal, message }: ErrorModalProps) => {
         <View style={styles.centeredView}>
           <View style={styles.modalView}>
             <View style={styles.modalCloseIconWrapper}>
-              <TouchableOpacity style={styles.modalCloseIcon}>
+              <TouchableOpacity style={styles.modalCloseIcon} onPress={() => setOpenModal(false)}>
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/ErrorModal.tsx]: fixed in a new commit

File: packages/app/src/hooks/useRecentTransactions.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/hooks/useRecentTransactions.ts]: ah wait.. ok now connecting the dots. this is related to the open question about 'how to / where to retrieve' recent transactions from

File: packages/app/src/components/StewardsList/StewardsListItem.tsx [hunk 12]
```diff
@@ -15,20 +16,21 @@ interface StewardListItemProps {
...
+  const { navigate } = useCrossNavigate();
 
   const isVerified = useIsStewardVerified(steward.steward);
 
   const { data: ensName } = useEnsName({ address: steward.steward as `0x${string}`, chainId: 1 });
   const userIdentifier = ensName ? ensName : formatAddress(steward.steward, 5);
 
   return (
-    <View style={styles.row}>
+    <TouchableOpacity style={styles.row} onPress={() => navigate(`/profile/${steward.steward}`)}>
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/StewardsList/StewardsListItem.tsx]: I don't think we need a new function for a one-line lambda

File: packages/app/src/components/TransactionList/TransactionList.tsx [hunk 13]
```diff
@@ -1,51 +1,41 @@
...
       </View>
-      {isDesktopResolution && (
-        <TouchableOpacity onPress={() => setShowAll(!showAll)} style={styles.showMoreButton}>
-          <Text style={styles.showMoreText}>Show {showAll ? 'less' : 'more'}</Text>
-          <Image
-            source={chevronDown}
-            style={[styles.showMoreIcon, { transform: [{ rotate: !showAll ? '0deg' : '180deg' }] }]}
-          />
+      {isDesktopResolution && transactions.length > 5 && (
+        <TouchableOpacity onPress={() => navigate('/profile/abc123/activity')} style={styles.showMoreButton}>
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/TransactionList/TransactionList.tsx]: same as above

File: packages/app/src/hooks/useContractCalls/useSupportFlowWithSwap.ts [hunk 14]
```diff
@@ -74,7 +74,11 @@ export function useSupportFlowWithSwap(
       return;
     } catch (error) {
       toggleCompleteDonationModal(false);
-      onError(`An unexpected error occurred: ${error}`);
+      console.error(error);
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/hooks/useContractCalls/useSupportFlowWithSwap.ts]: fixed in a new commit

File: packages/app/src/pages/HomePage.tsx [hunk 15]
```diff
@@ -64,6 +64,7 @@ const styles = StyleSheet.create({
     flexWrap: 'wrap',
     justifyContent: 'center',
     alignItems: 'flex-start',
+    gap: 24,
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/pages/HomePage.tsx:67]: @krisbitney fyi, not a style preference this comment, this actually breaks on native

[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/pages/HomePage.tsx:67]: I understood. I plan to fix it. Thanks.

[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/pages/HomePage.tsx:67]: After some research, I found that support for `gap` was added to React Native in version 0.71.0. I think we should be fine.

File: packages/app/src/components/DonateComponent.tsx [hunk 16]
```diff
@@ -156,6 +156,18 @@ function DonateComponent({ collective }: DonateComponentProps) {
 
   const { buttonCopy, buttonBgColor, buttonTextColor } = donateStyles;
 
+  const onChangeCurrency = (value: string) => setCurrency(value);
+  const onChangeAmount = (value: string) => setDecimalDonationAmount(formatDecimalStringInput(value));
+  const onChangeFrequency = (value: string) => {
+    if (currency !== 'G$' && value === Frequency.OneTime) {
```
Comment (L03TJ3) [inline packages/app/src/components/DonateComponent.tsx:162]: @krisbitney should this be a callback if you verify + set the currency?

File: packages/app/src/pages/HomePage.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/pages/HomePage.tsx:67]: The app is currently using `"react-native": "^0.71.11"`

File: packages/app/src/components/Breadcrumb.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/components/Breadcrumb.tsx]: thats just the standard event from an onPress: https://reactnative.dev/docs/pressevent

File: packages/app/src/components/DonateComponent.tsx (hunk 16 above)
Comment (krisbitney) [reply packages/app/src/components/DonateComponent.tsx]: Sure thing. I memoized it in a new commit.

File: packages/app/src/hooks/useRecentTransactions.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/hooks/useRecentTransactions.ts]: yes, exactly. This PR currently fixes the UI for the recent transactions list. This hook was created to decouple the UI from fetching the recent transactions. So now just the hook needs to be updated.

[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/hooks/useRecentTransactions.ts]: ok, lets fix that in separate PR

**Comment by krisbitney** [comment]
> @krisbitney could you in the description list with #issue which ones exactly are fixed (this helps in managing board on zenhub)

No, there are too many of them. Please just look at all the issues created in the last few days. The issue creation and QA process are GoodDollar's own internal processes, and I am not a GoodDollar employee.

This project is not what we call a "team extension" where GoodDollar would be paying me hourly. Please understand that I do not receive a salary from dOrg. I am receiving just $3,000 for all the work I did or will do on this project.

File: packages/app/src/components/Breadcrumb.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/Breadcrumb.tsx]: We are talking about a one-line method call that takes a constant value `-1`.

Please change this later if you want to track button clicks.

**Pull request merged**